code/modules/mob/living/simple_animal/abnormality/teth/penitent_girl.dm

```diff
@@ -25,38 +26,83 @@
 	gift_type =  /datum/ego_gifts/sorrow
 	abnormality_origin = ABNORMALITY_ORIGIN_WONDERLAB
 
-
+//Work Mechanics
 /mob/living/simple_animal/hostile/abnormality/penitentgirl/AttemptWork(mob/living/carbon/human/user, work_type)
-	//Temp too high, random damage type time.
-	if(get_attribute_level(user, TEMPERANCE_ATTRIBUTE) >= 40)
+	//Prudence too high, random damage type time.
+	if(get_attribute_level(user, PRUDENCE_ATTRIBUTE) >= 40)
 		work_damage_type = pick(WHITE_DAMAGE, RED_DAMAGE, BLACK_DAMAGE, PALE_DAMAGE)
 	return TRUE
 
 /mob/living/simple_animal/hostile/abnormality/penitentgirl/PostWorkEffect(mob/living/carbon/human/user, work_type, pe, work_time)
 	// you are going to cut your own leg off
-	if(get_attribute_level(user, TEMPERANCE_ATTRIBUTE) < 40)
-		user.apply_status_effect(STATUS_EFFECT_PENITENCE)
-		to_chat(user, "<span class='danger'>Something feels strange.</span>") //I have to write something here
+	if((get_attribute_level(user, TEMPERANCE_ATTRIBUTE) < 40) && (get_attribute_level(user, PRUDENCE_ATTRIBUTE) < 40))
+		user.apply_damage(250, WHITE_DAMAGE, null, user.run_armor_check(null, WHITE_DAMAGE), spread_damage = TRUE)//DIE!
 		work_damage_type = initial(work_damage_type)
 
+	if(user.sanity_lost)
+		user.apply_status_effect(STATUS_EFFECT_PENITENCE)
+		user.say("Why can't I have a little bit of freedom? Her shoes look perfect!")
+		QDEL_NULL(user.ai_controller)
+		user.ai_controller = /datum/ai_controller/insane/wander/penitence
+		user.InitializeAIController()
+		user.apply_status_effect(/datum/status_effect/panicked_type/wander/penitence)
```

Comment:
Is it not something that ai controller applies on its own?

Reply:
I just tested it, it's all required or weird things happen, like the old status effects persisting.